src/ir/function.rs

```diff
@@ -186,7 +186,8 @@ impl FunctionSig {
             }
         }
 
-        let ret = try!(Item::from_ty(&ty.ret_type(), None, None, ctx));
+        let ty_ret_type = ty.ret_type().expect("Expected a valid return type");
```

Comment:
This is not equivalent to the code that was before. If a type was invalid, before it'll generate a `Err(Continue)`, and the `try!` macro would make us return it earlier.

With this we're panicking, which isn't desirable.